### PR TITLE
Add import for `google_sql_database_instance`

### PR DESCRIPTION
**Pull request by gh-mlfowler**
Previously raised as https://github.com/hashicorp/terraform/pull/15235.

```
$ make testacc TESTARGS='-run=TestAccGoogleSqlDatabaseInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccGoogleSqlDatabaseInstance_ -timeout 120m
?       github.com/terraform-providers/terraform-provider-google        [no test files]
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic (62.66s)
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic3
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic3 (299.44s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic
--- PASS: TestAccGoogleSqlDatabaseInstance_basic (52.04s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic2
--- PASS: TestAccGoogleSqlDatabaseInstance_basic2 (63.26s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic3
--- PASS: TestAccGoogleSqlDatabaseInstance_basic3 (300.31s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_basic
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_basic (42.03s)
=== RUN   TestAccGoogleSqlDatabaseInstance_slave
--- PASS: TestAccGoogleSqlDatabaseInstance_slave (702.61s)
=== RUN   TestAccGoogleSqlDatabaseInstance_diskspecs
--- PASS: TestAccGoogleSqlDatabaseInstance_diskspecs (301.20s)
=== RUN   TestAccGoogleSqlDatabaseInstance_maintenance
--- PASS: TestAccGoogleSqlDatabaseInstance_maintenance (436.09s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_upgrade
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_upgrade (59.18s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_downgrade
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_downgrade (48.40s)
=== RUN   TestAccGoogleSqlDatabaseInstance_authNets
--- PASS: TestAccGoogleSqlDatabaseInstance_authNets (75.18s)
=== RUN   TestAccGoogleSqlDatabaseInstance_multipleOperations
--- PASS: TestAccGoogleSqlDatabaseInstance_multipleOperations (79.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 2521.860s```

### CODE REVIEW

File: google/resource_sql_database_instance.go [hunk 1]
```diff
@@ -638,7 +647,17 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
...
 
 	_settingsList := d.Get("settings").([]interface{})
-	_settings := _settingsList[0].(map[string]interface{})
+	var _settings map[string]interface{}
+
+	// If we're importing the settings will be nil
+	if len(_settingsList) > 0 {
+		_settings = _settingsList[0].(map[string]interface{})
+	} else {
+		_settingsList = make([]interface{}, 1)
```
Comment (gh-mlfowler) [inline google/resource_sql_database_instance.go]: Yes, at the end of the Read, https://github.com/gh-mlfowler/terraform-provider-google/blob/245bf4359c166b683eda35b81881fab076cccc34/google/resource_sql_database_instance.go#L850-L851

File: google/resource_sql_database_instance.go [hunk 2]
```diff
@@ -630,6 +634,11 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
+	// In the import case name won't be set yet
```
Comment (gh-mlfowler) [inline google/resource_sql_database_instance.go]: I've had to add a `d.setId` to the Create as the resource relied on it being set at the end of the first Read.

**Comment by gh-mlfowler** [comment]
Thanks for the review @rileykarson, I have made your suggested amendments.

```$ make testacc TESTARGS='-run=TestAccGoogleSqlDatabaseInstance_'                                                                                                                     
==> Checking that code complies with gofmt requirements...                                                                                       
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccGoogleSqlDatabaseInstance_ -timeout 120m                                      
?       github.com/terraform-providers/terraform-provider-google        [no test files]                                                          
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic                                                                                           
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic (61.91s)                                                                                  
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic3                                                                                          
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic3 (382.40s)                                                                                
=== RUN   TestAccGoogleSqlDatabaseInstance_basic                                                                                                 
--- PASS: TestAccGoogleSqlDatabaseInstance_basic (51.22s)                                                                                        
=== RUN   TestAccGoogleSqlDatabaseInstance_basic2                                                                                                
--- PASS: TestAccGoogleSqlDatabaseInstance_basic2 (51.78s)                                                                                       
=== RUN   TestAccGoogleSqlDatabaseInstance_basic3                                                                                                
--- PASS: TestAccGoogleSqlDatabaseInstance_basic3 (434.49s)                                                                                      
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_basic                                                                                        
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_basic (41.80s)                                                                               
=== RUN   TestAccGoogleSqlDatabaseInstance_slave                                                                                                 
--- PASS: TestAccGoogleSqlDatabaseInstance_slave (744.54s)                                                                                       
=== RUN   TestAccGoogleSqlDatabaseInstance_diskspecs                                                                                             
--- PASS: TestAccGoogleSqlDatabaseInstance_diskspecs (429.58s)                                                                                   
=== RUN   TestAccGoogleSqlDatabaseInstance_maintenance                                                                                           
--- PASS: TestAccGoogleSqlDatabaseInstance_maintenance (396.00s)                                                                                 
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_upgrade                                                                                      
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_upgrade (73.19s)                                                                             
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_downgrade                                                                                    
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_downgrade (62.82s)                                                                           
=== RUN   TestAccGoogleSqlDatabaseInstance_authNets                                                                                              
--- PASS: TestAccGoogleSqlDatabaseInstance_authNets (88.65s)                                                                                     
=== RUN   TestAccGoogleSqlDatabaseInstance_multipleOperations                                                                                    
--- PASS: TestAccGoogleSqlDatabaseInstance_multipleOperations (94.58s)                                                                           
PASS                                                                                                                                             
ok      github.com/terraform-providers/terraform-provider-google/google 2912.964s   ```

File: google/resource_sql_database_instance.go (hunk 1 above)
Comment (rileykarson) [inline google/resource_sql_database_instance.go]: It looks like when we use the  `_settingsList` slice down there, we are just overwriting it's contents with the `_settings` object. We can introduce a new slice to wrap `_settings` in there, and better semantically show that the two slices are not linked. That way, we can eliminate this line and break an implicit dependency between these parts of the function.

**Comment by gh-mlfowler** [comment]
```$ make testacc TESTARGS='-run=TestAccGoogleSqlDatabaseInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccGoogleSqlDatabaseInstance_ -timeout 120m
?       github.com/terraform-providers/terraform-provider-google        [no test files]
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic (50.59s)
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic3
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic3 (389.04s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic
--- PASS: TestAccGoogleSqlDatabaseInstance_basic (50.72s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic2
--- PASS: TestAccGoogleSqlDatabaseInstance_basic2 (49.15s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic3
--- PASS: TestAccGoogleSqlDatabaseInstance_basic3 (368.82s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_basic
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_basic (40.48s)
=== RUN   TestAccGoogleSqlDatabaseInstance_slave
--- PASS: TestAccGoogleSqlDatabaseInstance_slave (758.15s)
=== RUN   TestAccGoogleSqlDatabaseInstance_diskspecs
--- PASS: TestAccGoogleSqlDatabaseInstance_diskspecs (377.73s)
=== RUN   TestAccGoogleSqlDatabaseInstance_maintenance
--- PASS: TestAccGoogleSqlDatabaseInstance_maintenance (469.70s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_upgrade
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_upgrade (68.62s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_downgrade
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_downgrade (47.40s)
=== RUN   TestAccGoogleSqlDatabaseInstance_authNets
--- PASS: TestAccGoogleSqlDatabaseInstance_authNets (74.59s)
=== RUN   TestAccGoogleSqlDatabaseInstance_multipleOperations
--- PASS: TestAccGoogleSqlDatabaseInstance_multipleOperations (87.62s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 2832.630s```

File: google/resource_sql_database_instance.go (hunk 1 above)
Comment (gh-mlfowler) [inline google/resource_sql_database_instance.go]: I have made the amendments, please let me know if I understood correctly.

**Comment by rileykarson** [comment]
LGTM - thanks for the great work on this @gh-mlfowler!

```
TF_ACC=1 go test ./google -v -run=TestAccGoogleSqlDatabaseInstance_ -timeout 120m
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic (59.14s)
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic3
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic3 (454.68s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic
--- PASS: TestAccGoogleSqlDatabaseInstance_basic (59.49s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic2
--- PASS: TestAccGoogleSqlDatabaseInstance_basic2 (57.87s)
=== RUN   TestAccGoogleSqlDatabaseInstance_basic3
--- PASS: TestAccGoogleSqlDatabaseInstance_basic3 (375.44s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_basic
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_basic (37.98s)
=== RUN   TestAccGoogleSqlDatabaseInstance_slave
--- PASS: TestAccGoogleSqlDatabaseInstance_slave (831.64s)
=== RUN   TestAccGoogleSqlDatabaseInstance_diskspecs
--- PASS: TestAccGoogleSqlDatabaseInstance_diskspecs (352.84s)
=== RUN   TestAccGoogleSqlDatabaseInstance_maintenance
--- PASS: TestAccGoogleSqlDatabaseInstance_maintenance (393.30s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_upgrade
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_upgrade (55.26s)
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_downgrade
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_downgrade (45.28s)
=== RUN   TestAccGoogleSqlDatabaseInstance_authNets
--- PASS: TestAccGoogleSqlDatabaseInstance_authNets (60.85s)
=== RUN   TestAccGoogleSqlDatabaseInstance_multipleOperations
--- PASS: TestAccGoogleSqlDatabaseInstance_multipleOperations (83.13s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	2867.086s
```

**Pull request merged**